aten/src/ATen/native/cpu/Activation.cpp

```diff
@@ -175,11 +175,41 @@ void GeluBackwardKernelImpl(const Tensor& dY, const Tensor& X, Tensor* dX) {
   }
 }
 
+void hardshrink_cpu_kernel(TensorIterator& iter, Scalar lambd) {
+  AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "hardshrink_cpu", [&] {
+    auto lambd_val = lambd.to<scalar_t>();
+    cpu_kernel_vec(iter,
+      [=](scalar_t self_val) {
+        return (self_val >= -lambd_val && self_val <= lambd_val) ? scalar_t(0) : self_val;
+      },
+      [=](Vec256<scalar_t> self_val) {
+        return ((self_val < -lambd_val) | (self_val > lambd_val)) & self_val;
+      }
+    );
+  });
+}
+
+void hardshrink_backward_cpu_kernel(TensorIterator& iter, Scalar lambd) {
+  AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "hardshrink_backward_cpu", [&] {
+    auto lambd_val = lambd.to<scalar_t>();
+    cpu_kernel_vec(iter,
+      [=](scalar_t grad_val, scalar_t self_val) {
+        return (self_val >= -lambd_val && self_val <= lambd_val) ? scalar_t(0) : grad_val;
+      },
+      [=](Vec256<scalar_t> grad_val, Vec256<scalar_t> self_val) {
+        return ((self_val < -lambd_val) | (self_val > lambd_val)) & grad_val;
```

Comment:
Im not sure we support & for such use case, can you please write a test that compares internal implementation and manual python iteration over numel > 50000k 1D tensor.